### PR TITLE
docs update for multiple-onRequest feature

### PR DESCRIPTION
**Pull request by webbdays**

### CODE REVIEW

File: docs/directives.md [hunk 1]
```diff
@@ -973,6 +973,21 @@ type Post {
 
 - `query: {key: "id", value: "{{.value.userId}}"}]`: Instructs TailCall CLI to generate a URL aligning the user id with `userId` from the parent `Post`, compiling a single URL for a batch of posts, such as `/users?id=1&id=2&id=3...id=10`, consolidating requests into one.
 
+### onRequest
+
+A property which accepts string value and helps in defining middleware/filter for the requests. The middleware defined in http directive is applicable to only there. This only works when a js worker file is linked.
```
Comment (tusharmath) [inline docs/directives.md]: Add a link to the upstream setting also, stating that it can be overridden over here.

Comment (webbdays) [reply docs/directives.md]: ok.

Comment (webbdays) [reply docs/directives.md]: currently, we give preference to http directive right.
If not defined there, we check the upstream.

Comment (webbdays) [reply docs/directives.md]: // marge http and upstream on_request
            let http_filter = http
                .on_request
                .clone()
                .or(config_module.upstream.on_request.clone())
                .map(|on_request| HttpFilter { on_request });

**Review by tusharmath — APPROVED** [review]

**Pull request merged**